sdk/provisioning/Azure.Provisioning.AppContainers/tests/BasicAppContainersTests.cs

```diff
@@ -0,0 +1,172 @@
+                        Sku = new OperationalInsightsWorkspaceSku { Name = OperationalInsightsWorkspaceSkuName.PerGB2018 }
+                    };
+
+                ContainerAppManagedEnvironment env =
+                    new(nameof(env))
+                    {
+                        Location = location,
+                        AppLogsConfiguration =
+                            new ContainerAppLogsConfiguration
+                            {
+                                Destination = "log-analytics",
+                                LogAnalyticsConfiguration = new ContainerAppLogAnalyticsConfiguration
+                                {
+                                    CustomerId = logAnalytics.CustomerId,
+                                    SharedKey = logAnalytics.GetKeys().PrimarySharedKey,
+                                }
+                            },
+                    };
+
+                ContainerApp app =
+                    new(nameof(app))
+                    {
+                        Location = location,
+                        ManagedEnvironmentId = env.Id,
+                        Configuration =
+                            new ContainerAppConfiguration
+                            {
+                                Ingress =
+                                    new ContainerAppIngressConfiguration
+                                    {
+                                        External = true,
+                                        TargetPort = 80,
+                                        AllowInsecure = false,
+                                        Traffic =
+                                        {
+                                            new ContainerAppRevisionTrafficWeight
+                                            {
+                                                IsLatestRevision = true,
+                                                Weight = 100
+                                            }
+                                        }
+                                    },
+                            },
+                        Template =
+                            new ContainerAppTemplate
+                            {
+                                RevisionSuffix = "firstrevision",
+                                Scale = new ContainerAppScale { MinReplicas = 1, MaxReplicas = 3 },
+                                Containers =
+                                {
+                                    new ContainerAppContainer
+                                    {
+                                        Name = "test",
+                                        Image = containerImage,
+                                        Resources =
+                                            new AppContainerResources
+                                            {
+                                                Cpu = new BicepValue<double>(BicepFunction.ParseJson("0.5").Expression!),
```

Comment:
Or you can use expressions to solve fun problems like Bicep not supporting floating point numbers and telling people the correct way to provide them is using `json()` to parse float strings.  (I don't know why I added an explicit `BicepValue<double>` here though...  that's not strictly needed.)